dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/tests/TelemetryMetricCombinationTest.java

```diff
@@ -0,0 +1,103 @@
+
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.SERVER_ONLY;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+import com.ibm.websphere.microprofile.faulttolerance.metrics.fat.tests.isolation.IsolationServlet;
+import com.ibm.websphere.microprofile.faulttolerance.metrics.fat.tests.isolation.PullExporterAutoConfigurationCustomizerProvider;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.MicroProfileActions;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+
+/*
+ * This test ensures that the metrics exported to OpenTelemetry do not leak between two apps,
+ * and do not leak between the same application after it has been restarted.
+ */
+@RunWith(FATRunner.class)
+public class TelemetryMetricCombinationTest {
+
+    private static final String SERVER_NAME = "CDIFaultToleranceMetricsAndTelemetry";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests r = MicroProfileActions.repeat(SERVER_NAME, MicroProfileActions.MP70);
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        //We may as well reuse a pre-existing app since we're only testing a single metric, any metric
+        WebArchive combinationTestApp = ShrinkWrap.create(WebArchive.class, "CombinationTestAppOne.war")
+                        .addPackage(IsolationServlet.class.getPackage())
+                        .addAsManifestResource(MetricRemovalTest.class.getResource("removal/permissions.xml"), "permissions.xml")
+                        .addAsResource(new StringAsset("otel.sdk.disabled=false"),
+                                       "META-INF/microprofile-config.properties")
+                        .addAsServiceProvider(AutoConfigurationCustomizerProvider.class, PullExporterAutoConfigurationCustomizerProvider.class);
+
+        ShrinkHelper.exportAppToServer(server, combinationTestApp, SERVER_ONLY);
+
+        server.startServer();
+    }
+
+    @Test
+    public void metricTelemetryCombinationTest() throws Exception {
+        HttpUtils.findStringInUrl(server, "CombinationTestAppOne/isolationtest", "Test one passed Test two passed");
+
+        assertThat(getMetricsPage(), containsString("ft_retry_retries_total"));
+    }
```

Comment:
I think the first line calls a method and checks Telemetry reports it, and the second line checks MP Metrics reports it?

If so, please add a comment for each line.